src/com/facebook/buck/jvm/kotlin/KotlinTestDescription.java

```diff
@@ -89,65 +89,76 @@ public Arg createUnpopulatedConstructorArg() {
           new BuildTargetSourcePath(testTarget));
     }
 
-    BuildTarget abiJarTarget =
-        BuildTarget.builder(params.getBuildTarget())
-            .addFlavors(CalculateAbi.FLAVOR)
-            .build();
+    JavacOptions javacOptions =
+        JavacOptionsFactory.create(
+            templateJavacOptions,
+            params,
+            resolver,
+            pathResolver,
+            args
+        );
+
+    BuildTarget abiJarTarget = params.getBuildTarget().withAppendedFlavors(CalculateAbi.FLAVOR);
 
     KotlincToJarStepFactory stepFactory = new KotlincToJarStepFactory(
         kotlinBuckConfig.getKotlinCompiler().get(),
         args.extraKotlincArguments);
 
-    BuildRuleParams testsLibraryParams =
-        params.appendExtraDeps(
-            Iterables.concat(
-                BuildRules.getExportedRules(
+    BuildRuleParams testsLibraryParams = params.copyWithDeps(
+        Suppliers.ofInstance(
+            ImmutableSortedSet.<BuildRule>naturalOrder()
+                .addAll(params.getDeclaredDeps().get())
+                .addAll(BuildRules.getExportedRules(
                     Iterables.concat(
                         params.getDeclaredDeps().get(),
-                        resolver.getAllRules(args.providedDeps))),
-                pathResolver.filterBuildRuleInputs(
-                    defaultJavacOptions.getInputs(pathResolver))))
-            .withFlavor(JavaTest.COMPILED_TESTS_LIBRARY_FLAVOR);
-    JavaLibrary testsLibrary =
+                        resolver.getAllRules(args.providedDeps))))
+                .addAll(pathResolver.filterBuildRuleInputs(
+                    javacOptions.getInputs(pathResolver)))
+                .build()
+        ),
+        params.getExtraDeps())
+    .withFlavor(JavaTest.COMPILED_TESTS_LIBRARY_FLAVOR);
+
+    KotlinLibrary testsLibrary =
         resolver.addToIndex(
-            new DefaultJavaLibrary(
+            new DefaultKotlinLibrary(
                 testsLibraryParams,
                 pathResolver,
                 args.srcs,
                 ResourceValidator.validateResources(
                     pathResolver,
                     params.getProjectFilesystem(),
                     args.resources),
-                defaultJavacOptions.getGeneratedSourceFolderName(),
-                /* proguardConfig */ Optional.empty(),
-                /* postprocessClassesCommands */ ImmutableList.of(),
-                /* exportDeps */ ImmutableSortedSet.of(),
-                /* providedDeps */ ImmutableSortedSet.of(),
```

Comment:
They weren't consistently present for all parameters, and were not present in the JavaLibraryDescription.  We can add them back in; though if we do, they should probably appear after the parameters rather than before to make the code "prettier".